plugins/out_s3/s3.c

```diff
@@ -986,11 +987,19 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
     int size_check = FLB_FALSE;
     int part_num_check = FLB_FALSE;
     int timeout_check = FLB_FALSE;
-    time_t create_time;
     int ret;
     void *payload_buf = NULL;
     size_t payload_size = 0;
     size_t preCompress_size = 0;
+    time_t file_first_log_time = 0;
+
+    /*
+     * When chunk does not exist, file_first_log_time will be 0.
+     * This is only for unit tests.
+     */
+    if (chunk != NULL) {
+        file_first_log_time = chunk->first_log_time;
+    }
 
```

Comment:
I just want to note that we are now one small future code change away from having a bug here. 

There's a check in the cb_s3_flush: `if ((upload_file != NULL)`

Which ensures this code never runs in the normal execution case. 

But it does run for unit tests. 

But what if someone changes that code in the future?

Therefore, as we discussed verbally today I think it might really be better to have the `unit_test_flush` code create a chunk and set the timestamp on it. As @matthewfala said, we shouldn't modify the code for the unit tests.

Reply:
@Claych notes that this doesn't work since the code would try to delete the fake s3_file that is passed in.

Reply:
if we have chunk as NULL
then we could initialize first log time to time(NULL)
and then add a comment in the code that the if ((upload_file != NULL) is important, so please don't remove it
just to guard against future mistakes
Or may be am I being too over-protective here?